src/main/java/com/redhat/sast/api/service/EventBroadcastService.java

```diff
@@ -0,0 +1,87 @@
+package com.redhat.sast.api.service;
+
+import com.redhat.sast.api.event.BatchProgressEvent;
+import com.redhat.sast.api.event.JobStatusChangedEvent;
+import com.redhat.sast.api.event.OshScanCollectedEvent;
+import com.redhat.sast.api.event.OshScanFailedEvent;
+import com.redhat.sast.api.model.Job;
+import com.redhat.sast.api.model.JobBatch;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.event.Event;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Service for broadcasting real-time updates using CDI events.
+ *
+ * Fires CDI events that are observed by interested listeners such as:
+ * - WebSocketResource for dashboard updates
+ * - Logging/metrics services
+ * - Downstream workflow triggers
+ */
+@ApplicationScoped
+@RequiredArgsConstructor
+@Slf4j
+public class EventBroadcastService {
+
+    private final Event<JobStatusChangedEvent> jobStatusChangedEvent;
```

Comment:
I prefer your earlier implementation which we kept the sessions in the ws resource class and notifying all the clients. It was simple. With this approach, it feels we are making things overcomplicated. wdyt?